extensions/provisioning/ssh.go

```diff
@@ -22,53 +24,72 @@ import (
-// For example CheckCPU checks the cpu usage of the cluster agent. If the usage is too high the func will return a warning.
 func CallSSHTestByName(testCase provisioninginput.SSHTestCase, node *nodes.Node, client *rancher.Client, clusterID string, machineName string) error {
 	switch testCase {
+	//checks the cpu usage of all processes on the node. If the usage is too high the function will return a warning.
 	case checkCPU:
 		logrus.Infof("Running CheckCPU test on node %s", node.PublicIPAddress)
 		output, err := node.ExecuteCommand(checkCPUCommand)
-		if err != nil {
-			return err
+		if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
+			return errors.New(err.Error() + output)
 		}
-		strOutput := output[:strings.IndexByte(output, '\n')]
-		logrus.Info("CheckCPU test on node " + node.PublicIPAddress + " | Cluster agent cpu usage is: " + strOutput + "%")
 
-		outputInt, err := strconv.ParseFloat(strings.TrimSpace(strOutput), 32)
-		if outputInt > cpuUsageVar {
-			logrus.Warn("Cluster agent cpu usage is too high on node" + node.PublicIPAddress + " | Current cpu usage is: " + strOutput + "%")
-		}
-		if err != nil {
-			return err
+		lines := strings.Split(output, "\n")
+		logrus.Info("Checking all node processes CPU usage")
+		for _, line := range lines {
+			processFields := strings.Fields(line)
+			if len(processFields) > 0 {
+				CPUUsageInt, err := strconv.ParseFloat(strings.TrimSpace(processFields[1]), 32)
+				if err != nil {
+					return errors.New(err.Error() + output)
+				}
+
+				if CPUUsageInt >= cpuUsageTolerance {
+					logrus.Warnf("Process: %s | CPUUsage: %f", processFields[0], CPUUsageInt)
+				}
+			}
 		}
+
+	//This test reboots the node and verifies it comes back up in the correct state.
 	case nodeReboot:
 		logrus.Infof("Running NodeReboot test on node %s", node.PublicIPAddress)
-		command := "sudo reboot"
-		_, err := node.ExecuteCommand(command)
+		output, err := node.ExecuteCommand(rebootNodeCommand)
 		if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
-			return err
+			return errors.New(err.Error() + output)
 		}
-		// Verify machine shuts down within five minutes, shutting down should not take longer than that depending on the ami
-		err = wait.Poll(1*time.Second, defaults.FiveMinuteTimeout, func() (bool, error) {
-			newNode, err := client.Steve.SteveType(machineSteveResourceType).ByID(fleetNamespace + "/" + machineName)
-			if err != nil {
-				return false, err
-			}
-			if newNode.State.Name == runningState {
-				return false, nil
-			}
-			return true, nil
-		})
+
+		err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, defaults.FiveMinuteTimeout, true,
+			func(ctx context.Context) (done bool, err error) {
+				newNode, err := client.Steve.SteveType(machineSteveResourceType).ByID(fleetNamespace + "/" + machineName)
+				if err != nil {
+					return false, err
+				}
+
+				if newNode.State.Name == runningState {
+					return false, nil
+				}
+				return true, nil
```

Comment:
nit; please put this on a new line for code readability.

Reply:
I'll be running our linter against this today once i get it running in vscode and it will catch that